packages/template-retail-react-app/app/hooks/use-item-movement.js

```diff
@@ -0,0 +1,252 @@
+     * @returns {Promise<Object>} The updated basket response
+     */
+    const moveItemToPickupShipment = useCallback(
+        async (productItem, targetShipmentId, inventoryId) => {
+            if (!basketId || !productItem?.itemId) {
+                throw new Error('Invalid basket or product item')
+            }
+
+            try {
+                // Update the item to add inventory ID and move to pickup shipment
+                const updateData = {
+                    productId: productItem.productId,
+                    quantity: productItem.quantity,
+                    shipmentId: targetShipmentId,
+                    inventoryId: inventoryId
+                }
+
+                return await updateItemInBasketMutation.mutateAsync({
+                    parameters: {
+                        basketId,
+                        itemId: productItem.itemId
+                    },
+                    body: updateData
+                })
+            } catch (error) {
+                handleError('Failed to move item to pickup shipment', error)
+                throw error
+            }
+        },
+        [basketId, updateItemInBasketMutation, handleError]
+    )
+
+    /**
+     * Moves a product item from pickup to delivery shipment
+     * @param {Object} productItem - The product item to move
+     * @param {string} targetShipmentId - The target shipment ID (optional)
+     * @param {string} defaultInventoryId - The default inventory ID to use for delivery items (required)
+     * @returns {Promise<Object>} The updated basket response
+     */
+    const moveItemToDeliveryShipment = useCallback(
+        async (productItem, targetShipmentId = DEFAULT_SHIPMENT_ID, defaultInventoryId) => {
+            if (!basketId || !productItem?.itemId) {
+                throw new Error('Invalid basket or product item')
+            }
+
+            try {
+                // Update the item to remove inventory ID and move to different shipment
+                const updateData = {
+                    productId: productItem.productId,
+                    quantity: productItem.quantity,
+                    shipmentId: targetShipmentId
+                }
+
+                // Set inventoryId to default for delivery items (instead of null which doesn't work)
+                if (productItem.inventoryId) {
+                    updateData.inventoryId = defaultInventoryId
+                }
+
+                return await updateItemInBasketMutation.mutateAsync({
+                    parameters: {
+                        basketId,
+                        itemId: productItem.itemId
+                    },
+                    body: updateData
+                })
+            } catch (error) {
+                handleError('Failed to move item to delivery shipment', error)
+                throw error
+            }
+        },
+        [basketId, updateItemInBasketMutation, handleError]
```

Comment:
I'd advise think carefully about passing these functions as dependencies
What does `updateItemInBasketMutation`, `handleError` dependency mean here? How would it change that it can trigger this callback?